### PR TITLE
Create api ping filter

### LINKED ISSUE #186: Create API ping filter
Create a filter which adds the API ping to the image, which acts as a rough estimate for the latency the participant experiences

### PR DESCRIPTION
**Pull request by BorderBoy**
# Overview
This PR adds a ping filter, which displays the current API ping value in the bottom left corner.
The ping is fetched using the filter API which was extended.

# Changes
- Added ping filter
- Added `_send_command_wait_for_response()` to `FilterSubprocessAPI` which makes it possible to send a command to the main process and wait for a response. This was only possible in the other direction before.
- Added functionality to user class which sends pings to frontend and calculates avg. ping
- Added ping messages which go from server to client
- Rename `server_time` in PONG message to `handled_time` so the name fits regardless in which direction its send.

There are some parameters which can be adjusted by the experimenter:
- Frequency of PING signals which are sent: currently 1/s
- Length of buffer (moving avg): currently: 30s

## Wiki-Changes
- https://github.com/TUMFARSynchrony/experimental-hub/wiki/_compare/89c6c6f38bceb6950a198d983a901aa1df6fb011
- https://github.com/TUMFARSynchrony/experimental-hub/wiki/_compare/7e489f6d62639da8951cc54caab3b9472bc3ab7a

# How To Test?
Create a new session and add a PING filter.
There are also some minor frontend changes, so do not forget to `npm run build` when testing.

# Future Work
Display ping directly in frontend, instead of on the image (like #164)

Closes #186

### CODE REVIEW

**Comment by NCEghtebas** [comment]
Do we want to merge this now or wait to include the displaying from PR #164?

**Comment by BorderBoy** [comment]
Ok I just noticed a race condition which could lead to closing the connection. I need to fix this before we can merge it.

**Comment by BorderBoy** [comment]
@NCEghtebas I made the parameters configurable. So this should now be ready to merge if it's reviewed

**Review by NCEghtebas — APPROVED** [review]

**Pull request merged**
